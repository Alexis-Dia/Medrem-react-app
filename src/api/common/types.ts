import { ActionType } from "typesafe-actions";
import { AddressShortDto } from "../../types/address";
import { Language } from "../../types/common";
import {
    AttachmentType,
    Competitor,
    ContactType,
    ContractDocumentTypeDto,
    ContractStatus,
    ContractType,
    CustomerType,
    DeliveryAccess,
    DeliveryToDo,
    DeliveryType,
    DocumentType,
    Interval,
    OfferType,
    RejectionReason,
} from "../../types/contract";
import { DeliveryAddressItemType } from "../../types/product";
import { SettingType } from "../../types/settings";
import { SystemInformationType } from "../../types/systemInformation";
import * as commonActions from "./commonActions";
import { CalculationDto } from "../../types/common/CalculationDto";

export interface SaveSuggestionPayload {
    key: string;
    value?: string | AddressShortDto;
    index?: number;
}

export interface CommonState {
    offerStatuses: Array<ContractStatus>;
    orderStatuses: Array<ContractStatus>;
    availableStatuses: Array<ContractStatus>;
    userRoles: Array<string>;
    userFeatures: Array<string>;
    rectifications: Array<Interval>;
    offerTypes: Array<OfferType>;
    documentTypes: Array<DocumentType>;
    deliveryTypes: Array<DeliveryType>;
    deliveryToDos: Array<DeliveryToDo>;
    deliveryAccesses: Array<DeliveryAccess>;
    contractTypes: Array<ContractType>;
    customerTypes: Array<CustomerType>;
    contactTypes: Array<ContactType>;
    calculations: Array<CalculationDto>;
    billingIntervals: Array<Interval>;
    featuresForRoles: Array<string>;
    settingsTypes: Array<SettingType>;
    languages: Array<Language>;
    attachmentTypes: Array<AttachmentType>;
    itemTypes: Array<DeliveryAddressItemType>;
    suggestionStatus: SaveSuggestionPayload | null;
    systemInformationTypes: Array<SystemInformationType>;
    rejectionReasons: Array<RejectionReason>;
    competitors: Array<Competitor>;
    officeBranchIds: Array<number>;
    contractDocumentTypes: Array<ContractDocumentTypeDto>;
}

export interface CommonState {
    offerStatuses: Array<ContractStatus>;
    orderStatuses: Array<ContractStatus>;
    availableStatuses: Array<ContractStatus>;
    userRoles: Array<string>;
    userFeatures: Array<string>;
    rectifications: Array<Interval>;
    offerTypes: Array<OfferType>;
    documentTypes: Array<DocumentType>;
    deliveryTypes: Array<DeliveryType>;
    deliveryToDos: Array<DeliveryToDo>;
    deliveryAccesses: Array<DeliveryAccess>;
    contractTypes: Array<ContractType>;
    customerTypes: Array<CustomerType>;
    contactTypes: Array<ContactType>;
    calculations: Array<CalculationDto>;
    billingIntervals: Array<Interval>;
    featuresForRoles: Array<string>;
    settingsTypes: Array<SettingType>;
    languages: Array<Language>;
    attachmentTypes: Array<AttachmentType>;
    itemTypes: Array<DeliveryAddressItemType>;
    suggestionStatus: SaveSuggestionPayload | null;
    systemInformationTypes: Array<SystemInformationType>;
    rejectionReasons: Array<RejectionReason>;
    competitors: Array<Competitor>;
    officeBranchIds: Array<number>;
    contractDocumentTypes: Array<ContractDocumentTypeDto>;
}

export type CommonAction = ActionType<typeof commonActions>;
