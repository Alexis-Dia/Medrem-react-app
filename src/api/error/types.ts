import { ActionType } from "typesafe-actions";
import { ErrorDto } from "../../types/error/ErrorDto";
import * as errorActions from "./errorActions";

export type ErrorState = ErrorDto | null;

export type ErrorAction = ActionType<typeof errorActions>;
