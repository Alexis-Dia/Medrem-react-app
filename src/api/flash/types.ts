import { ActionType } from "typesafe-actions";
import { FlashMessage } from "../../types/flash";
import * as flashActions from "./flashActions";

export type FlashMessagesState = Array<FlashMessage>;

export type FlashMessagesAction = ActionType<typeof flashActions>;
