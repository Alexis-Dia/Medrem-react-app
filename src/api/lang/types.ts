import { ActionType } from "typesafe-actions";
import * as localeActions from "./localeActions";

export interface LocaleState {
    lang: string;
}

export type LocaleAction = ActionType<typeof localeActions>;
