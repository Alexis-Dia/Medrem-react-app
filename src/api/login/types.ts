import { ActionType } from "typesafe-actions";
import { Language } from "../../types/common";
import * as loginActions from "./loginActions";

export interface UserErrors {
    invalidCredentials: string;
    serverIsUnreachable: string;
}

export interface UserPayload {
    username: string | null;
    password: string | null;
    errors: UserErrors | string;
    originErrors?: string;
}

export interface LogOutPayload {
    originLogOutMessage: string;
    logOutMessage: string;
}

interface HandshakeDto {
    token: string;
	features: string;
	roles: string;
	username: string;
	identity: string;
	language: Language;
}

interface HandshakeDto {
    token: string;
	features: string;
	roles: string;
	username: string;
	identity: string;
	language: Language;
}

export interface LoginState {
    isAuthenticated: boolean,
    onCloseClicked: boolean,
    user: Partial<HandshakeDto> & Partial<UserPayload>;
}

export type LoginAction = ActionType<typeof loginActions>;
