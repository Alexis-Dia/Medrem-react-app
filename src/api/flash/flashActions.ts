import { createAction } from "typesafe-actions";

export const addFlashMessage = createAction(
    "ADD_FLASH_MESSAGE",
    (type: string, text: string) => ({ type, text }),
)();

export const addAutoRemovableFlashMessage = createAction(
    "ADD_AUTO_REMOVABLE_FLASH_MESSAGE",
    (type: string, text: string, delay: number) => ({ type, text, delay }),
)();

export const deleteFlashMessageByText = createAction(
    "DELETE_FLASH_MESSAGE_BY_TEXT",
    (text: string) => ({ text }),
)();

export const deleteFlashMessageById = createAction(
    "DELETE_FLASH_MESSAGE_BY_ID",
    (id: string) => ({ id }),
)();

export const deleteAllFlashMessages = createAction(
    "DELETE_ALL_FLASH_MESSAGES",
)();
