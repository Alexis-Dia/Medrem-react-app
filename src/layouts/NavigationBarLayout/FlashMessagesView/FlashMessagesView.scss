.flash-message-main-style {
  position: absolute;
  z-index: 10000;
  width: 25%;
  top: 90px;
  right: 40px;
  word-wrap: break-word;
}
