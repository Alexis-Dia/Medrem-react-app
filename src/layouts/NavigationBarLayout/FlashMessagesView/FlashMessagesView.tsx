import React from "react";
import { connect, ConnectedProps } from "react-redux";
import { bindActionCreators } from "redux";
import { deleteFlashMessageById } from "../../../api/flash/flashActions";
import { RootState } from "../../../store/reducers";
import FlashMessageView from "../FlashMessage/FlashMessage";
import "./FlashMessagesView.scss";

type Props = PropsFromRedux;

const FlashMessagesView = (props: Props) => (
    <div className="flash-message-main-style">
        {props.flashMessages.map((flashMessage) => (
            <FlashMessageView
                key={flashMessage.id}
                type={flashMessage.type}
                text={flashMessage.text}
                onCloseClick={() => props.deleteFlashMessageById(flashMessage.id)}
            />
        ))}
    </div>
);

const mapStateToProps = (state: RootState) => ({
    flashMessages: state.flashMessages,
});

const mapDispatchToProps = (dispatch: any) => ({
    deleteFlashMessageById: bindActionCreators(deleteFlashMessageById, dispatch),
});

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(FlashMessagesView);
