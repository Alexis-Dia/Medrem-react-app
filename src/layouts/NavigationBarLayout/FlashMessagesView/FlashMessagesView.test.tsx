import React from "react";
import configureStore from "redux-mock-store";
import FlashMessagesView from "./FlashMessagesView";
import { getWrappedComponent, mountComponent, renderComponent, Wrapper } from "../../../utils/test-utils";
import { FlashMessage, FlashMessageType } from "../../../types/flash";
import FlashMessageView from "../FlashMessage/FlashMessage";
import { deleteFlashMessageById } from "../../../api/flash/flashActions";

const mockStore = configureStore();
const store = mockStore({
    flashMessages: [
        { id: 5, type: FlashMessageType.SUCCESS, text: "success-message" },
        { id: 11, type: FlashMessageType.ERROR, text: "error-message" }
    ]
});

const getWrappedComponentImpl = () => {
    const wrappers = [Wrapper.REDUX_PROVIDER];
    const component = <FlashMessagesView />
    return { component: getWrappedComponent(component, wrappers, {}, store), name: "FlashMessagesView" };
};

jest.mock("../FlashMessage/FlashMessage", () => "flash-message-mock");

describe("<FlashMessagesView />", () =>  {
    const { wrapper: wrapperWithInitialProps, component: componentWithInitialProps } = mountComponent(getWrappedComponentImpl());

    it("renders correctly", () => {
        const tree = renderComponent(getWrappedComponentImpl()).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should pass props to children", () => {
        const { flashMessages } = store.getState();

        flashMessages.forEach((flashMessage: FlashMessage, index: number) => {
            const flashMessageComponent = wrapperWithInitialProps.find(FlashMessageView).at(index);
            expect(flashMessageComponent.key()).toBe(String(flashMessage.id));

            const flashMessageComponentProps = flashMessageComponent.props();
            expect(flashMessageComponentProps.type).toBe(flashMessage.type);
            expect(flashMessageComponentProps.text).toBe(flashMessage.text);
        });
    });

    it("clicks on flash message close icon", () => {
        const flashMessageComponent = wrapperWithInitialProps.find(FlashMessageView).first();
        flashMessageComponent.props().onCloseClick();

        const { id } = store.getState().flashMessages[0];
        expect(store.getActions()).toContainEqual(deleteFlashMessageById(id));
    });
});