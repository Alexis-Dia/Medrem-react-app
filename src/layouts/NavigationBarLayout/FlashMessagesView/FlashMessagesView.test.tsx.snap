// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashMessagesView /> renders correctly 1`] = `
<div
  className="flash-message-main-style"
>
  <flash-message-mock
    onCloseClick={[Function]}
    text="success-message"
    type="success"
  />
  <flash-message-mock
    onCloseClick={[Function]}
    text="error-message"
    type="error"
  />
</div>
`;
